workers/exclusive-content/README.md

```diff
@@ -0,0 +1,92 @@
+# Exclusive Content Worker
+
+This cloudflare worker will encrypt any plain text given a payment pointer. Furthermore, it will decrypt the resulting cypher text if it can verify payment by a receipt verifier.
+
+## Testing
+
+Configure your `wrangler.toml` file.
+
+```sh
+npm install
+wrangler secret put MASTERKEY
+# enter a key to be stored with cloudflare worker
+wrangler preview --watch
+```
+
+## Deploying
+
+Configure your `wrangler.toml` file.
+
+```sh
+npm install
+wrangler secret put MASTERKEY
+# enter a key to be stored with cloudflare worker
+wrangler publish
+```
+
+## Endpoints
+
+### `POST /encrypt`
+
+Encrypts plaintext
+
+#### Request Body
+
+JSON object containing
+| Name | Type | Description | Example |
+| -- | -- | -- | -- |
+| pp | String | (proxy) payment pointer | "\$spsp.example.com/alice" |
+| pt | String | plain text | "Hello World" |
+
+#### Headers
+
+| Header       | Value            |
+| ------------ | ---------------- |
+| Content-Type | application/json |
+
+#### Response Body
+
+JSON object containing
+| Name | Type | Description | Example |
+| -- | -- | -- | -- |
+| ct | String | cypher text | "VsPD81db5Ts9rUnvxqGD5hxCRoDU+ygE44w42AoEbaKQbhGXkrpxoYwCrhrtR2MmFfOg" |
+| iv | String | initialization vector | "1603118464654" |
+
+#### Errors
+
+- 400 - Input variables missing
+- 415 - Unsupported content type. Use application/json
+
+### `POST /decrypt`
+
+Decrypts cyphertext if it can verify payment
+
+#### Request Body
+
+JSON object containing
+| Name | Type | Description | Example |
+| -- | -- | -- | -- |
+| pp | String | (proxy) payment pointer | "\$spsp.example.com/alice" |
+| ct | String | cypher text | "VsPD81db5Ts9rUnvxqGD5hxCRoDU+ygE44w42AoEbaKQbhGXkrpxoYwCrhrtR2MmFfOg" |
+| iv | String | initialization vector | "1603118464654" |
+| vr | String | receipt verifier endpoint | "https://verifier.example.com" |
```

Comment:
I can pass in any endpoint I like here and have it say that I paid. Should probably be an env var to the worker instead. Another option would be to specify the verifier at encryption time and include it in the ciphertext (you could encrypt a json object to accomodate several fields) so that it can't be tampered with

Reply:
Yes, you are right. I prefer option 2 since that way everybody can use their own verifier. But for a first implementation, we could also use one verifier for everybode (the new service). 
If we are moving encryption and decryption of the plaintext out of the worker, the verifier decryption would still happen inside the worker but I guess that is ok.

Reply:
yeah if we're not passing the ciphertext to the server it's a little tricky. For this version the most reasonable solution is probably just to configure a verifier endpoint as a variable for the worker.

Another option would be to have a JWT-like signed token that the exclusive content generator produces at encryption time and takes in at decryption time. But maybe that's a feature for later if we can get away with having one verifier for now

Reply:
I actually went with the flow of passing the encrypted verifier endpoint but I'm happy to change that.